http-client/src/main/java/io/airlift/http/client/jetty/JettyResponseFuture.java

```diff
@@ -70,6 +74,19 @@ public String getState()
         return state.get().toString();
     }
 
+    @Override
+    public void onCommit(org.eclipse.jetty.client.Request request)
+    {
+        state.set(JettyAsyncHttpState.REQUEST_COMMITED);
+    }
+
+    public void idleTimeout()
+    {
+        if (state.get() == JettyAsyncHttpState.WAITING_FOR_CONNECTION) {
```

Comment:
Since this can be called from a different thread than the other Jetty state handling, we probably need to make all the methods synchronized. Otherwise, for example, we could be in the middle of executing `cancel()` and have set the status to cancelled but not updated the state yet. Also, this is just failing the future, but isn't touching the Jetty state, so `cancel()` or `completed()` could be called after this.

This idle timeout change seems to be more complicated than it appears. Why do we need to do this ourselves? Does the Jetty client already do this and maybe we're missing a callback or something?

How is it related to the HTTP/2 change?

If this change is required in the current form, then it will take some careful consideration of the response future state machine. Let's pull this out into a separate PR.